### DISCUSSION TITLE
Diff two solutions on two meshes

### DISCUSSION THREAD

**Discussion by tangqi** [mfem/mfem#1933] [Q&A] [answered]
Is there a simple way to diff two solutions on two non conforming meshes? For instance, one is a refined solution on a uniform mesh and one is an AMR solution. I found the field-diff app. The example is very simple, but will that work in parallel AMR meshes (quad)? Thanks.

If that's not possible, other suggestions are also welcome. I am trying to do convergence study for problems with no exact solutions.

Qi

**Comment by vladotomov** [comment, marked as answer]
Hi Qi,

Yes `field-diff` should work with AMR meshes, but we don't have a parallel version yet. You can use the `PrintAsOne` and `SaveAsOne` methods of `ParMesh` and `ParGridFunction` and use `field-diff` on the resulting files.

Vladimir

**Comment by tangqi** [comment]
Hi, I try to compare a full mesh vs an amr mesh. It is not working for me. See the attached solutions. 
[archive.tar.gz](https://github.com/mfem/mfem/files/5967215/archive.tar.gz)

Thanks,
Qi

PS, the amr is non-conforming but it is saved as "MFEM mesh v1.0". But I do not think it is the reason.

**Reply by tangqi** [reply]
ping @vladotomov @tzanio

**Reply by vladotomov** [reply]
Hi Qi,

Thank you for reporting. I fixed some things and was able to run your comparison with the branch gslib-fix, the PR is #2052.

<img width="1792" alt="Screen Shot 2021-02-12 at 10 19 00 PM" src="https://user-images.githubusercontent.com/17843342/107843700-188cf880-6d82-11eb-8abb-df107937d59d.png">

**Reply by tangqi** [reply]
Vladimir, thanks for the quick fix. It looks great!
